runtime/sema/type.go

```diff
@@ -6448,6 +6518,33 @@ func (t *AddressType) initializeMemberResolvers() {
 	})
 }
 
+func IsPrimitiveOrContainerOfPrimitive(ty Type) bool {
+	if ty.IsPrimitiveType() {
+		return true
+	}
+
+	// TODO: Do we also want to count Dictionary?
```

Comment:
Yes! Dictionaries should also be considered containers the same way arrays are

Reply:
So the main question is, do we want `Dictionary<PrimitiveKey, Non-PrimitiveValue>` or `Dictionary<Non-Primitive key, Primitive value` to also support `dereference`?

I guess we only want `Dictionary<primitive key, primitive value>` to support dereference but wanted to confirm.

Reply:
Dictionary keys are already restricted to be subtypes of "Hashable Structs" (see `IsHashableStructType`), which similar to the Primitive type designation (and might actually be able to be refactored to use the new `IsPrimitiveType` method). Point is though, that I don't think you need to worry about the key types at all when checking whether a dictionary reference is dereference-able, just check the values.

Reply:
Done